### DISCUSSION TITLE
Does top languages count forks?

### DISCUSSION THREAD

**Discussion by Metacinnabar** [anuraghazra/github-readme-stats#1255] [Q&A] [answered]
Do the languages used in forks count towards the top language?

**Comment by Hephaest** [comment, marked as answer]
Dear @GoodPro712, the answer is **NOT**. You can check the corresponding API callback:
```JavaScript
const fetcher = (variables, token) => {
  return request(
    {
      query: `
      query userInfo($login: String!) {
        user(login: $login) {
          # fetch only owner repos & not forks
          repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
            nodes {
              name
              languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
                edges {
                  size
                  node {
                    color
                    name
                  }
                }
              }
            }
          }
        }
      }
      `,
      variables,
    },
    {
      Authorization: `bearer ${token}`,
    },
  );
};
```
You can check the source code by clicking this [link](https://github.com/anuraghazra/github-readme-stats/blob/master/src/fetchers/top-languages-fetcher.js).

### Additional information
If you want to explore more or play a little bit with this filter query, please read this official [Github GraphQL Document](https://docs.github.com/en/graphql/reference/interfaces#repositoryinfo).

**Comment by rickstaa** [comment]
Pasting my comment in https://github.com/anuraghazra/github-readme-stats/pull/1122#issuecomment-1152066225 since it explains why forks are currently not supported.

> Since the current [GraphQL languages object](https://docs.github.com/en/graphql/reference/objects#language) returns the languages used in a repository, not those in the commits done by a specific user, the current version of the PR will make the language stats less accurate. I created a feature request to improve this behaviour on GitHub's side (see https://github.com/orgs/github-community/discussions/18230). You can show your support there.
